thresher/serializers.py

```diff
@@ -199,14 +199,87 @@ def to_internal_value(self, data):
                                
 
 # A serializer for a highlight group
-class HighlightGroupSerializer(serializers.ModelSerializer):
-    # a custom field containing all the questions and answers
-    questions = serializers.ListField(child=GenericSubmittedAnswerField())
+# class HighlightGroupSerializer(serializers.ModelSerializer):
+#     # a custom field containing all the questions and answers
+#     questions = serializers.ListField(child=GenericSubmittedAnswerField())
+#     offsets = JSONSerializerField()
+
+#     class Meta:
+#         model = HighlightGroup
+#         fields = ('tua', 'offsets', 'questions')
+
+#     def create(self, validated_data):
+#         # Get the answers nested models
+#         answers = validated_data.pop('questions')
+
+#         # Remove the force_insert if it's there
+#         validated_data.pop('force_insert', None)
+
+#         # create the highlight group first
+#         highlight_group = HighlightGroup.objects.create(**validated_data)
+
+#         # Add the highlight group model to the kwargs and save
+#         for answer in answers:
+#             model = answer['class']
+#             kwargs = answer['data']
+#             # Add the highlight group instance to the kwargs
+#             kwargs['highlight_group'] = highlight_group
+            
+#             # There is a special case if it's a checklist,
+#             # Because of the many to many relationship, this needs to be saved differently
+#             if model == CLSubmittedAnswer:
+#                 # Get the answers:
+#                 answers = kwargs.pop('answer')
+#                 # first create the CLSubmitted answer
+#                 submission = CLSubmittedAnswer.objects.create(**kwargs)
+#                 # Now add the answers
+#                 submission.answer.add(*answers)
+
+#             # For all other models, simply create the objects
+#             else:
+#                 model.objects.create(**kwargs)
+
+#         return highlight_group 
+
+class OffsetField(serializers.Field):
```

Comment:
@p-patil Are the errors still an issue? If so could you post what they are here?